### DISCUSSION TITLE
Adding multiple classes with `x-bind`

### DISCUSSION THREAD

**Discussion by khoivan88** [alpinejs/alpine#1555] [7. Bugs (archived)] [answered]
Hi, 

First of all, thank you for this library. I have been using Alpine for a few projects of mine and I really enjoyed it. 

Yesterday, when I tried to upgrade the Alpine from 2.8.2 to 3.0.6, I think there is a bug with multiple classes added by `x-show`.

Test: current working mobile menu with Alpine 2.8.2: https://kristufeklab.com/
This is with Alpine 3.0 and even with me fixing the `x-init`, adding the multiple classes and remove a specific class does not work: https://deploy-preview-67--kristufeklab.netlify.app/

One of the problematic code blocks is the highlighted one in this screenshot:
![image](https://user-images.githubusercontent.com/33493502/122578765-344f2680-d022-11eb-8e33-6f74cd9039d2.png)

I would appreciate it if anyone would have a pointer for me to see if this would work on Alpine 3. I tried for several hours yesterday to no avail.

Thank you very much!

**Comment by KevinBatdorf** [comment]
Probably related to this?

https://github.com/alpinejs/alpine/discussions/1525#discussioncomment-885384

**Reply by khoivan88** [reply]
Yes, I saw that Discussion 'room' yesterday but did not see that last comment (https://github.com/alpinejs/alpine/discussions/1525#discussioncomment-885384). It seems that the last comment is almost exactly the same situation I am in. 
The earlier discussion said that he had some typo in the coding so I was not sure if it would apply to my case here!
Thanks!

**Reply by danddanddand** [reply]
Ya, I suspect it's same problem. Search your code for :class="{ 'a' : isOpen, 'b': !isOpen } and replace with ternary :class="isOpen ? 'a' : 'b'" for now as work around.

**Reply by khoivan88** [reply]
Thank you for your suggestion, unfortunately, I don't think that would work in my case since I already have some classes and I want to add a few more and remove one when the button is clicked! Of course, I can create a custom class but that seems to defeat the purpose of using Alpine!

I thought this function was really powerful and hopefully can be implemented in v3. I was not sure if I had missed some special syntax or something!

**Reply by danddanddand** [reply]
Okay, we had to hack a temp solution you can try until more complicated cases are supported again properly with object syntax. You could try:

:class="
                                            [
                                                yourConditionA ? 'border border-blue-400' : '',
                                                yourConditionB === focusedOptionIndex ? 'bg-gray-200' : 'text-gray-900',
                                                yourConditionC(options.indexOf( option ) !== focusedOptionIndex) ? 'bg-blue-50' : ''
                                            ].join( ' ' )
                                        "

**Reply by khoivan88** [reply]
If I have some time this weekend, I will give that a try, I do need to translate that into my situation block code first. Thank you very much for all of your suggestions 😄  👍

**Reply by danddanddand** [reply]
Ya, it was just pseudocode but basically you're just collecting list of classes you want :class="[ isBlah? 'blah' : '', isFoo? 'foo': '', isBar? 'bar' : ''].join( ' ' )" ... not very elegant but essentially does what v2 would've done with :class="{'blah': isBlah, 'foo': isFoo, 'bar': isBar }"

**Reply by khoivan88** [reply]
learning some new way to write code today, so cool! Thanks @danddanddand

**Reply by danddanddand** [reply]
> learning some new way to write code today, so cool! Thanks @danddanddand

There was a fix merged so I assume when next alpine release it tagged your old code would work again.

**Reply by khoivan88** [reply]
That would be amazing. Thank you very much!

**Reply by KevinBatdorf** [reply]
He tagged the release if you could check

**Reply by khoivan88** [reply]
Hi @KevinBatdorf ,

I tested it again with Alpine 3.0.8 and it still not working as in v2. I am not sure if I am missing anything!

**Reply by KevinBatdorf** [reply]
Can you update the URL above? It still shows 3.0.6. Or even better if you could recreate it on Codepen

**Reply by khoivan88** [reply]
Hi @KevinBatdorf, sorry for the delay, I was creating the simplified version on Codepen: https://codepen.io/khoivan88/pen/ExWJgyq

The problem is in medium or small screen, the hamburger menu does not open the actual menu. This line `<div class="top-0 left-0 items-start w-full h-full p-4 text-sm lg:items-center lg:w-5/6 lg:absolute lg:text-base lg:bg-transparent lg:p-0 lg:relative lg:flex hidden" :class="{'flex fixed': showMenu, 'hidden': !showMenu }">` would have work on alpine v2 to display the menu.

Thank you very much for having a look!

**Reply by KevinBatdorf** [reply]
That issue is different. You need to stop the event from your button from bubbling up so that the `outside` listener doesn't trigger. Do this on the button:

```
@click.stop="showMenu = !showMenu"
```

**Reply by khoivan88** [reply]
@KevinBatdorf worked like a charm. Thank you so much!!
Would you mind giving me a pointer on why this was not an issue with Alpine v2 but a problem on v3? I am quite a newbie with Alpine

**Reply by KevinBatdorf** [reply]
It may have been a bug in V2. Not sure. The click event is happening on the SVG and bubbling up each element until it reaches the window, in which case the `outside/away` listener is picking it up and firing its callback. 

By the way, the click event should be on a button not a div. If it must be a div, then add `role="button"` to it

**Reply by khoivan88** [reply]
I see. This is totally an error on my part. Thank you so much for your help looking over my code!

**Comment by khoivan88** [comment, marked as answer]
> That issue is different. You need to stop the event from your button from bubbling up so that the `outside` listener doesn't trigger. Do this on the button:
> 
> ```
> @click.stop="showMenu = !showMenu"
> ```

The answer by @KevinBatdorf
